lib/Logger.js

```diff
@@ -55,7 +77,9 @@ module.exports = class Logger {
 
   _output(level, record) {
     this._transports.forEach((transport) => {
-      transport(level, record);
+      const p = transport(level, record) || Promise.resolve();
+      this._promises.add(p);
+      p.then(() => this._promises.delete(p));
```

Comment:
I'm not sure what the performance implications of maintaining the set of promises. I do know it won't make tripitaka faster though. Hopefully it will be fast enough.